tests/performance/rewrite_sumIf.xml

```diff
@@ -0,0 +1,4 @@
+<test>
+    <query>SELECT sumIf(1, 0) FROM numbers(100000000)</query>
```

Comment:
Should flag `optimize_rewrite_sum_if_to_count_if` be enabled to compare new behavior with old? Also it is good to add functional test, to ensure that it works correctly.

Reply:
The flag is enabled default, and we can see the difference:
https://clickhouse-test-reports.s3.yandex.net/17041/1f7b9053311febfbbaa17b71cb265e6a9705b117/performance_comparison/report.html

Ok, i will add a functional test.